File: internal/pkg/database/models/directory_groups_sub_groups.go
package models

import (
	"github.com/gofrs/uuid/v5"
)

type DirectoryGroupsSubGroups struct {
	ParentGroupID []uuid.UUID `json:"parent_group_id,omitempty"`
	SubGroupID    []uuid.UUID `json:"sub_group_id,omitempty"`
}

type directoryGroupsSubGroupsTable struct {
	TableName string

	ParentGroupID string
	SubGroupID    string
}

func DirectoryGroupsSubGroupsTable() directoryGroupsSubGroupsTable {
	return directoryGroupsSubGroupsTable{
		TableName: "directory_groups_sub_groups",

		ParentGroupID: "parent_group_id",
		SubGroupID:    "sub_group_id",
	}
}

File: web/src/lib/database/directory_groups_sub_groups.ts
namespace _DirectoryGroupsSubGroups {
	export enum Column {
		ParentGroupID = 'parent_group_id',
		SubGroupID = 'sub_group_id'
	}

	export const TableName = 'directory_groups_sub_groups'

	export interface Interface {
		parent_group_id?: string[]
		sub_group_id?: string[]
	}
}

export default _DirectoryGroupsSubGroups
